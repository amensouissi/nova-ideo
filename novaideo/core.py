import colander
import deform.widget
from zope.interface import implementer

from substanced.util import renamer

from dace.objectofcollaboration.entity import Entity
from dace.descriptors import (
    SharedUniqueProperty,
    CompositeUniqueProperty,
    SharedMultipleProperty,
    CompositeMultipleProperty)
from dace.util import getSite, allSubobjectsOfKind
from pontus.schema import Schema
from pontus.core import VisualisableElement

from novaideo import _
from novaideo.content.interface import (
    IVersionableEntity,
    IDuplicableEntity,
    ISearchableEntity,
    ICommentable,
    ICorrelableEntity)


@implementer(ICommentable)
class Commentabl(VisualisableElement, Entity):
    name = renamer()
    comments = CompositeMultipleProperty('comments')


@implementer(IVersionableEntity)
class VersionableEntity(Entity):

    version = CompositeUniqueProperty('version', 'nextversion')
    nextversion = SharedUniqueProperty('nextversion', 'version')

    @property
    def current_version(self):
        if self.nextversion is None:
            return self
        else:
            return self.nextversion.current_version

    @property
    def history(self):
        return allSubobjectsOfKind(interface=IVersionableEntity)


@implementer(IDuplicableEntity)
class DuplicableEntity(Entity):

    originalideas = SharedMultipleProperty('originalideas')


@colander.deferred
def keywords_choice(node, kw):
    values = []
    root = getSite()
    prop = sorted(root.keywords, key=lambda p: p.title)
    values = [i.title for i in prop]
    return deform.widget.AutocompleteInputWidget(values=values)


@colander.deferred
def default_keywords_choice(node, kw):
    context = node.bindings['context']
    root = getSite()
    if context is root:
        return []

    values = sorted(context.keywords)
    return values


class SearchableEntitySchema(Schema):

    keywords =  colander.SchemaNode(
        colander.Sequence(),
        colander.SchemaNode(
             colander.String(),
             widget=keywords_choice,
             name='keyword'),
        default=default_keywords_choice,
        title=_('Keywords'),
        )


@implementer(ISearchableEntity)
class SearchableEntity(Entity):
    result_template = 'novaideo:templates/views/default_result.pt'
    keywords_ref = SharedMultipleProperty('keywords_ref', 'referenced_elements')

    @property
    def keywords(self):
        return [k.title for k in self.keywords_ref]


@implementer(ICorrelableEntity)
class CorrelableEntity(Entity):
    source_correlations = SharedMultipleProperty('source_correlations', 'source')
    target_correlations = SharedMultipleProperty('target_correlations', 'target')

    @property
    def correlations(self):
        result = [c.target for c in self.source_correlations]
        result.extend([c.source for c in self.target_correlations])
        return list(set(result))
