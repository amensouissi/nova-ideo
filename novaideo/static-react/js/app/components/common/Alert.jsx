import React from 'react';
import { withStyles } from 'material-ui/styles';
import classNames from 'classnames';

import { iconAdapter } from '../../utils/globalFunctions';

const styles = (theme) => {
  return {
    container: {
      padding: 16,
      border: '1px solid #e8e8e8',
      background: '#fff',
      color: '#2c2d30',
      borderLeftWidth: 5,
      borderRadius: 6,
      display: 'flex',
      '&.info': {
        borderLeftColor: theme.palette.info[500]
      },
      '&.danger': {
        borderLeftColor: theme.palette.danger.primary
      },
      '&.warning': {
        borderLeftColor: theme.palette.warning[500]
      },
      '&.success': {
        borderLeftColor: theme.palette.success[500]
      }
    },
    icon: {
      marginRight: 10,
      fontSize: '22px !important',
      marginTop: -7,
      '&.info': {
        color: theme.palette.info[500]
      },
      '&.danger': {
        color: theme.palette.danger.primary
      },
      '&.warning': {
        color: theme.palette.warning[500]
      },
      '&.success': {
        color: theme.palette.success[500]
      }
    }
  };
};

export const ALERTS_ICONS = {
  info: iconAdapter('mdi-set mdi-alert-circle-outline'),
  danger: iconAdapter('mdi-set mdi-alert-outline'),
  warning: iconAdapter('mdi-set mdi-alert-box'),
  success: iconAdapter('mdi-set mdi-check-circle-outline')
};

export const DumbAlert = ({ type = 'info', classes, children }) => {
  const Icon = ALERTS_ICONS[type];
  return (
    <div className={classNames(classes.container, type)}>
      <Icon className={classNames(classes.icon, type)} />
      <div>
        {children}
      </div>
    </div>
  );
};

export default withStyles(styles, { withTheme: true })(DumbAlert);