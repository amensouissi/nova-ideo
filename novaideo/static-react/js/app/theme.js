import { teal, grey, red, orange, blue, green } from '@material-ui/core/colors';
import { createMuiTheme } from '@material-ui/core/styles';

const primaryCode = 500;

// function theme() {
//   return createMuiTheme({
//     palette: {
//       primary: {
//         ...teal,
//         [primaryCode]: '#4D394B', // test color to remove.
//         light: '#cac4c9',
//         dark: '#342032',
//         dark2: '#2a1628'
//       },
//       secondary: grey,
//       tertiary: {
//         color: '#4C9689',
//         hover: {
//           color: 'white'
//         }
//       },
//       danger: red,
//       info: blue,
//       warning: orange
//     },
//     typography: {
//       htmlFontSize: 15,
//       fontFamily: '"LatoWebMedium", "Helvetica Neue", Helvetica, Arial, sans-serif'
//     },
//     body1: {
//       margin: 0
//     }
//   });
// }

function theme() {
  return createMuiTheme({
    palette: {
      primary: {
        ...teal,
        [primaryCode]: '#282e3f', // test color to remove.
        light: '#cac4c9',
        dark: '#1e222d',
        dark2: '#12151f'
      },
      secondary: grey,
      tertiary: {
        color: '#de9100',
        hover: {
          color: 'white'
        }
      },
      danger: {
        ...red,
        primary: '#d72b3f'
      },
      info: blue,
      warning: orange,
      success: green
    },
    typography: {
      htmlFontSize: 15,
      fontFamily: '"LatoWebMedium", "Helvetica Neue", Helvetica, Arial, sans-serif'
    },
    body1: {
      margin: 0
    }
  });
}

export default theme;