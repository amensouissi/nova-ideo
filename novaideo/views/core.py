# Copyright (c) 2015 by Ecreall under licence AGPL terms
# avalaible on http://www.gnu.org/licenses/agpl.html

# licence: AGPL
# author: Amen Souissi, Sophie Jazwiecki

from pontus.view import BasicView

from novaideo import _


class ActionAnonymousView(BasicView):
    title = _('Alert for comment')
    template = 'novaideo:views/templates/alert_action_anonymous.pt'

    def update(self):
        self.execute(None)
        result = {}
        values = {'context': self.context}
        body = self.content(args=values, template=self.template)['body']
        item = self.adapt_item(body, self.viewid)
        result['coordinates'] = {self.coordinates: [item]}
        return result


class ComponentView(BasicView):
    title = ''
    template = 'novaideo:views/templates/component_view.pt'
    wrapper_template = 'pontus:templates/views_templates/simple_view_wrapper.pt'
    component_id = ''

    def update(self):
        self.execute(None)
        result = {}
        body = self.content(
            args={'id': self.component_id},
            template=self.template)['body']
        item = self.adapt_item(body, self.viewid)
        result['coordinates'] = {self.coordinates: [item]}
        return result
