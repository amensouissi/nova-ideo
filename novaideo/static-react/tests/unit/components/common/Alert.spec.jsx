import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { createRender } from '@material-ui/core/test-utils';
import { MuiThemeProvider } from '@material-ui/core/styles';

import Alert from '../../../../js/app/components/common/Alert';
import theme from '../../../../js/app/theme';

configure({ adapter: new Adapter() });

describe('Alert component', () => {
  it('should render a danger alert', () => {
    const renderer = createRender();
    const component = renderer(
      <MuiThemeProvider theme={theme}>
        <Alert dismissible type="danger">
          Foo Bar
        </Alert>
      </MuiThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });
});