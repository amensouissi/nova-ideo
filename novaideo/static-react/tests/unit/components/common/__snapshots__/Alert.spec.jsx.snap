// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component should render a danger alert 1`] = `
<div
  class="DumbAlert-container-1 danger"
>
  <span
    aria-hidden="true"
    class="material-icons MuiIcon-root-4 mdi-set mdi-alert-outline DumbAlert-icon-2 danger"
  />
  <div>
    Foo Bar
  </div>
  <button
    class="MuiButtonBase-root-19 MuiIconButton-root-13 DumbAlert-closeBtn-3 danger"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label-18"
    >
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root-4 mdi-set mdi-close"
      />
    </span>
    <span
      class="MuiTouchRipple-root-22"
    />
  </button>
</div>
`;
